### PR TITLE
Issue309 bestest hydronic resources

### PR DESCRIPTION
**Pull request by JavierArroyoBastida**
This is for https://github.com/ibpsa/project1-boptest/issues/309. 
@dhblum this is NOT to be merged yet. I'll be waiting for https://github.com/ibpsa/project1-boptest/pull/310 before updating references here.

### CODE REVIEW

**Comment by dhblum** [comment]
#310 is merged.

**Comment by JavierArroyoBastida** [comment]
@dhblum now this one should be ready if tests pass. Could you have a look and merge when you get the chance?

File: testcases/bestest_hydronic/models/generate_data.py [hunk 1]
```diff
@@ -153,5 +153,54 @@
 # Add CO2 limits
 df['UpperCO2[1]'] = 894.
 
+def hold_step_values_in_event_times(df, var, event_freq):
+    ''' This method goes through the index of a data frame to inspect
+    when step changes take place for a variable and makes sure that
```
Comment (dhblum) [inline testcases/bestest_hydronic/models/generate_data.py:158]: Maybe just clarify as "positive" step changes?

File: testcases/bestest_hydronic_heat_pump/models/generate_data.py [hunk 2]
```diff
@@ -155,5 +155,53 @@
 # Add CO2 limits
 df['UpperCO2[1]'] = 894.
 
+def hold_step_values_in_event_times(df, var, event_freq):
+    ''' This method goes through the index of a data frame to inspect
+    when step changes take place for a variable and makes sure that
```
Comment (dhblum) [inline testcases/bestest_hydronic_heat_pump/models/generate_data.py:160]: Maybe just clarify as "positive" step changes?

**Review by dhblum — CHANGES_REQUESTED** [review]
Looks good in general, thanks for the fixes.  Just see one inline comment (same comment on two files).

File: testcases/bestest_hydronic/models/generate_data.py (hunk 1 above)
Comment (JavierArroyoBastida) [reply testcases/bestest_hydronic/models/generate_data.py]: Yep! clarified in https://github.com/ibpsa/project1-boptest/pull/311/commits/ed5d3b8e41f2e14d4f0a23e731e27e5c60a845c0.

File: testcases/bestest_hydronic_heat_pump/models/generate_data.py (hunk 2 above)
Comment (JavierArroyoBastida) [reply testcases/bestest_hydronic_heat_pump/models/generate_data.py]: And here: https://github.com/ibpsa/project1-boptest/pull/311/commits/184e08b3b3dbe426adf791460ebc01f9cadc2569.

File: testcases/bestest_hydronic/models/generate_data.py (hunk 1 above)
Comment (dhblum) [reply testcases/bestest_hydronic/models/generate_data.py]: Ok thanks.  Will merge after checks pass one last time.

**Review by dhblum — APPROVED** [review]

**Pull request merged**